File: src/WorkBC.Data/Model/JobBoard/JobAlert.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using WorkBC.Data.Enums;

namespace WorkBC.Data.Model.JobBoard
{
    public class JobAlert
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [ForeignKey("JobSeeker")]
        public string AspNetUserId { get; set; }

        public virtual JobSeeker JobSeeker { get; set; }

        [MaxLength(50)]
        public string Title { get; set; }

        [Column(TypeName = "tinyint")]
        public JobAlertFrequency AlertFrequency { get; set; }

        [MaxLength(1000)]
        public string UrlParameters { get; set; }

        public string JobSearchFilters { get; set; }
        public int JobSearchFiltersVersion { get; set; } = 1;
        public DateTime DateCreated { get; set; } = DateTime.Now;
        public DateTime? DateModified { get; set; }
        public DateTime? DateDeleted { get; set; }
        public bool IsDeleted { get; set; } = false;
    }
}

File: jobboard-laravel/app/Models/JobAlert.php
<?php

namespace App\Models;

use App\Models\Casts\TolerantEnum;
use App\Models\Concerns\SoftDeletesByFlag;
use App\Models\Enums\AlertFrequency;
use App\Search\Filters\JobSearchFiltersCast;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

final class JobAlert extends Model
{
    use SoftDeletesByFlag;

    protected $table = 'JobAlerts';

    protected $primaryKey = 'Id';

    protected $keyType = 'int';

    public $incrementing = true;

    public $timestamps = false;

    protected $casts = [
        'AlertFrequency' => TolerantEnum::class.':'.AlertFrequency::class,
        'JobSearchFilters' => JobSearchFiltersCast::class,
        'JobSearchFiltersVersion' => 'integer',
        'IsDeleted' => 'boolean',
        'DateCreated' => 'datetime',
        'DateModified' => 'datetime',
        'DateDeleted' => 'datetime',
    ];

    public function jobSeeker(): BelongsTo
    {
        return $this->belongsTo(JobSeeker::class, 'AspNetUserId', 'Id');
    }
}
